src/renderer/components/swap/Swap.tsx

```diff
@@ -332,6 +332,9 @@ export const Swap = ({
 
   const setAmountToSwap = useCallback(
     (targetAmount: BaseAmount) => {
+      // make sure the target amount decimal is the same as source asset decimal.
+      targetAmount.decimal = sourceAssetAmount.decimal
```

Comment:
Imagine: 
- source: ETH (18 decimal)
- target: BTC (8 decimal)

With this change ^ BTC will have 18 decimal ...

Reply:
Here, the targetAmount is the user input for source asset.

Reply:
Ah, I've missed that ^ ...